packages/muelu/research/regionMG/src/SetupRegionMatrix_def.hpp

```diff
@@ -122,283 +122,13 @@ Teuchos::Array<int> findCommonRegions(const GlobalOrdinal nodeA, ///< GID of fir
-          if  (jthRegions[i] == curRegion) break;
-        }
-        // (*regionsPerGIDWithGhosts)[0] entries keep original GID
-        // while others use firstNewGID to determine NewGID
-
-        if (j == 0) revisedGIDs.push_back(colGIDsComp[i]);
-        else if (j < appData.maxRegPerGID) {
-          revisedGIDs.push_back((int) firstNewGIDWithGhost->getData(0)[i] + j - 1);
-        }
-      }
-    }
-    revisedColMapPerGrp[k] = MapFactory::Build(AComp->getMap()->lib(),
-                                               Teuchos::OrdinalTraits<GlobalOrdinal>::invalid(),
-                                               revisedGIDs,
-                                               Teuchos::OrdinalTraits<GlobalOrdinal>::zero(),
-                                               AComp->getMap()->getComm());
-  }
-  for (std::size_t k = myRegions.size(); k < Teuchos::as<std::size_t>(appData.maxRegPerProc); k++) {
-    revisedRowMapPerGrp[k] = MapFactory::Build(AComp->getMap()->lib(),
-                                               Teuchos::OrdinalTraits<GlobalOrdinal>::invalid(),
-                                               Teuchos::OrdinalTraits<GlobalOrdinal>::zero(),
-                                               AComp->getMap()->getComm());
-    revisedColMapPerGrp[k] = MapFactory::Build(AComp->getMap()->lib(),
-                                               Teuchos::OrdinalTraits<GlobalOrdinal>::invalid(),
-                                               Teuchos::OrdinalTraits<GlobalOrdinal>::zero(),
-                                               AComp->getMap()->getComm());
-  }
-} // MakeGroupExtendedMaps
-
 template <class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
 void MakeQuasiregionMatrices(const RCP<Xpetra::CrsMatrixWrap<Scalar, LocalOrdinal, GlobalOrdinal, Node> > AComp,
-                             const int maxRegPerProc,
                              RCP<Xpetra::MultiVector<LocalOrdinal, LocalOrdinal, GlobalOrdinal, Node> > regionsPerGIDWithGhosts,
-                             std::vector<RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > >& rowMapPerGrp,
-                             std::vector<RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > >& colMapPerGrp,
-                             std::vector<RCP<Xpetra::Import<LocalOrdinal, GlobalOrdinal, Node> > >& rowImportPerGrp,
-                             std::vector<RCP<Xpetra::Matrix<Scalar, LocalOrdinal, GlobalOrdinal, Node> > >& quasiRegionGrpMats) {
+                             RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> >& rowMapPerGrp,
+                             RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> >& colMapPerGrp,
+                             RCP<Xpetra::Import<LocalOrdinal, GlobalOrdinal, Node> >& rowImportPerGrp,
+                             RCP<Xpetra::Matrix<Scalar, LocalOrdinal, GlobalOrdinal, Node> >& quasiRegionGrpMats) {
```

Comment:
The removed `std::vector` was related to the concept of "groups". Hence, this could be renamed to 
```suggestion
                             RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> >& rowMap,
                             RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> >& colMap,
                             RCP<Xpetra::Import<LocalOrdinal, GlobalOrdinal, Node> >& rowImport,
                             RCP<Xpetra::Matrix<Scalar, LocalOrdinal, GlobalOrdinal, Node> >& quasiRegionMats) {
```